src/read.c

```diff
@@ -799,6 +799,16 @@ static avifResult avifMetaFindOrCreateItem(avifMeta * meta, uint32_t itemID, avi
         }
     }
 
+    uint32_t prevID = 0;
+    // ISO/IEC 23008-12, First edition, 2017-12, Section 9.3.1:
+    //   Each ItemPropertyAssociation box shall be ordered by increasing item_ID, and there shall
+    //   be at most one association box for each item_ID, in any ItemPropertyAssociation box.
+    for (uint32_t i = 0; i < meta->items.count; ++i) {
+        AVIF_CHECKERR(meta->items.item[i]->id > prevID, AVIF_RESULT_BMFF_PARSE_FAILED);
+        prevID = meta->items.item[i]->id;
+    }
+    AVIF_CHECKERR(itemID > prevID, AVIF_RESULT_BMFF_PARSE_FAILED);
```

Comment:
If this is the only place where we add new items to `meta->items`, then we can assume the function maintains this property as an invariant, and the only validation we need to check is that `itemID` be greater than the last item ID in `meta->items`:

```
    if (meta->items.count != 0) {
        const uint32_t lastID = meta->items.item[meta->items.count - 1]->id;
        // Also log a diagnostic message!
        AVIF_CHECKERR(itemID > lastID, AVIF_RESULT_BMFF_PARSE_FAILED);
    }
```

Reply:
Indeed, let's keep it simple.